src/utils.py

```diff
@@ -94,8 +95,12 @@ def uin(param, default=0xDEADBEEF):
 
 
 def confirm(message):
-    print(message)
-    print('press Enter to confirm (ctrl-c to quit)')
+    if config.BUILDER_NON_INTERACTIVE:
+        errcho('Non-interactive mode, confirming automatically')
```

Comment:
lacks context without the `message` being printed. I would suggest using the logger in non-interactive mode and have these as debug statements

Reply:
a `logger` should be a standard way to use `stderr`

Reply:
yeeeessss, I agree on principle. but to get the same functionality with a logger one has to create a special handler with a special formatter. it's not the first thought that comes to mind when you 'just need to print to stderr'.

Reply:
Able to add a logger, but will use `info` as a level so that the output is included in the Jenkins logs